### PR TITLE
Whisker annotation

### LINKED ISSUE #2352: Feature Request: Error Bar Plots
For scientists one hugely important feature would be plots with error bars und lower/upper limits like these:

http://matplotlib.org/examples/statistics/errorbar_limits.html

Also see this stackexchange question:
http://stackoverflow.com/questions/29166353/how-do-you-add-error-bars-to-bokeh-plots-in-python

### PR DESCRIPTION
**Pull request by canavandl**
- [x] issues: fixes #2352 
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by canavandl** [comment]
looks like (could perhaps use some better defaults):

screen shot 2017-06-04 at 8 17 23 pm

**Comment by bryevdv** [comment]
Just an idea: What about adding a `TeeHead` for arrows, then implementing this in terms on an Arrow internally? Then users could swap out different end things if they wanted and all the drawing code ould be reused

**Comment by canavandl** [comment]
> Just an idea: What about adding a TeeHead for arrows, then implementing this in terms on an Arrow internally? Then users could swap out different end things if they wanted and all the drawing code ould be reused

That's a good idea, i'll make the change.

**Comment by bryevdv** [comment]
I guess `plus_end` and `minus_end` would allow users to control the sides independently, with the interpration of `base + thing` and `base - thing` to define which side is which in a way independent of dimensions? Alternatively just provide one head that is always used for both ends.

**Comment by bryevdv** [comment]
I also wonder if this ought to default to a different level? I know we have an "annotations" level under the assumption that most annotations will want to be "on top" but I feel like maybe whiskers are usually underneath? not sure

File: bokehjs/src/coffee/models/annotations/whisker.coffee [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      _upper_vx = limit_scale.v_compute(@_upper)
+    else
+      _upper_vx = @_upper
+
+    if @model.base.units == "data"
+      _base_vx = base_scale.v_compute(@_base)
+    else
+      _base_vx = @_base
+
+    @_canvas = @plot_model.canvas
```
Comment (mattpap) [inline bokehjs/src/coffee/models/annotations/whisker.coffee]: This should be only a local variable.

File: bokehjs/src/coffee/models/annotations/whisker.coffee [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    else
+      _lower_vx = @_lower
+
+    if @model.upper.units == "data"
+      _upper_vx = limit_scale.v_compute(@_upper)
+    else
+      _upper_vx = @_upper
+
+    if @model.base.units == "data"
+      _base_vx = base_scale.v_compute(@_base)
```
Comment (mattpap) [inline bokehjs/src/coffee/models/annotations/whisker.coffee]: Where did `@_base` came from? The same thing happens in `Band` annotation.

Comment (mattpap) [inline bokehjs/src/coffee/models/annotations/whisker.coffee]: Never mind, `base` is a spec, so `_base` is magically created in `materialize_dataspecs()`.

**Comment by canavandl** [comment]
Final:
screen shot 2017-06-05 at 2 46 50 pm

**Comment by bryevdv** [comment]
I think the example would look a little better with an alpha on the circles, but I'll be making a separate PR tomorrow for a pass over the docs, I can play around with it then. Looks good will merge when green

File: bokeh/models/annotations.py [hunk 3]
```diff
@@ -983,3 +983,57 @@ class Tooltip(Annotation):
...
+    upper_head = Instance('.models.arrow_heads.ArrowHead', default=_DEFAULT_TEE, help="""
+    Instance of ArrowHead.
+    """)
+
+    base = DistanceSpec(help="""
+    The orthogonal coordinates of the upper and lower values.
+    """)
+
+    dimension = Enum(Dimension, default='height', help="""
+    The direction of the band.
```
Comment (bryevdv) [inline bokeh/models/annotations.py]: Will update this in follow on docs PR

**Pull request merged**